### PR TITLE
use lazy user in UserMountCache

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Review by CarlSchwan — APPROVED** [review]

File: lib/private/Files/Config/UserMountCache.php [hunk 1]
```diff
@@ -204,7 +205,7 @@ private function removeFromCache(ICachedMountInfo $mount) {
 	}
 
 	private function dbRowToMountInfo(array $row) {
-		$user = $this->userManager->get($row['user_id']);
+		$user = new LazyUser($row['user_id'], $this->userManager);
 		if (is_null($user)) {
```
Comment (CarlSchwan) [inline lib/private/Files/Config/UserMountCache.php:209]: user is suddently never null anymore

**Review by CarlSchwan — CHANGES_REQUESTED** [review]
see last comment

File: lib/private/Files/Config/UserMountCache.php (hunk 1 above)
Comment (icewind1991) [reply lib/private/Files/Config/UserMountCache.php]: reworked the check

File: lib/private/Files/Config/UserMountCache.php [hunk 2]
```diff
@@ -204,10 +205,14 @@ private function removeFromCache(ICachedMountInfo $mount) {
 	}
 
 	private function dbRowToMountInfo(array $row) {
-		$user = $this->userManager->get($row['user_id']);
-		if (is_null($user)) {
+		$userid = $row['user_id'];
+
+		// check that the user exits
+		if ($this->userManager->getDisplayName($userid) === null) {
```
Comment (skjnldsv) [inline lib/private/Files/Config/UserMountCache.php:211]: Not fond of this check, seems hack-ish, no?

Comment (icewind1991) [reply lib/private/Files/Config/UserMountCache.php:211]: A bit yes, but it's the easiest way I can think of to do this check without requesting data from the user backend if not needed

Comment (icewind1991) [reply lib/private/Files/Config/UserMountCache.php:211]: https://github.com/nextcloud/server/pull/33582 add caching to `userExists`, will adjust this PR once that is merged

**Pull request closed without merging**